src/daft-core/src/series/ops/merge.rs

```diff
@@ -0,0 +1,77 @@
+use arrow2::offset::Offsets;
+use common_error::{DaftError, DaftResult};
+use daft_schema::{dtype::DataType, field::Field};
+
+use crate::{
+    array::{growable::make_growable, ListArray},
+    series::{IntoSeries, Series},
+};
+
+impl Series {
+    /// Merges series into a single series of lists.
+    /// ex:
+    /// ```text
+    /// A: Series := ( a_0, a_1, .. , a_n )
+    /// B: Series := ( b_0, b_1, .. , b_n )
+    /// C: Series := MERGE(A, B) <-> ( [a_0, b_0], [a_1, b_1], [a_2, b_2] )
+    /// ```
+    pub fn merge(field: Field, series: &[&Self]) -> DaftResult<Self> {
```

Comment:
Tagging @jaychia to please review this function 🙏

Reply:
Nit: maybe `zip` instead since it's common terminology for iterators?